backend/tests/test_api.py

```
@@ -387,22 +387,320 @@ def test_index_endpoint(self, mock_add_task, mock_load_config):
+
+    @patch('backend.api.filedialog')
+    @patch('backend.api.tk')
+    def test_browse_folder_success(self, mock_tk, mock_filedialog):
+        """Test folder browsing success."""
+        mock_root = MagicMock()
+        mock_tk.Tk.return_value = mock_root
+        mock_filedialog.askdirectory.return_value = '/selected/folder'
+
+        response = self.client.get("/api/browse")
+
+        self.assertEqual(response.status_code, 200)
```

Ruff (0.15.2)

[warning] Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)